### PR TITLE
Add linode

### PR DESCRIPTION
**Pull request by eyberg**

### CODE REVIEW

File: provider/linode/linode_store.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
+package linode
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
```
Comment (rinor) [inline provider/linode/linode_store.go:9]:
```suggestion
	"io"
```

File: provider/linode/linode_store.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+token)
+
+	res, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
```
Comment (rinor) [inline provider/linode/linode_store.go:52]:
```suggestion
	body, err := io.ReadAll(res.Body)
```

File: provider/linode/linode_store.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+func uploadImage(uri string, archPath string) {
+	f, err := os.Open(archPath)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	reader := bufio.NewReader(f)
+	content, err := ioutil.ReadAll(reader)
```
Comment (rinor) [inline provider/linode/linode_store.go:78]:
```suggestion
	content, err := io.ReadAll(reader)
```

File: provider/linode/linode_store.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	req.Header.Set("Content-Length", slen)
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer res.Body.Close()
+
+	_, err = ioutil.ReadAll(res.Body)
```
Comment (rinor) [inline provider/linode/linode_store.go:104]:
```suggestion
	_, err = io.ReadAll(res.Body)
```

File: provider/linode/linode.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	//	ctx := context.Background()
+	return nil
+}
+
+// GetStorage returns storage interface for cloud provider
+func (v *Linode) GetStorage() lepton.Storage {
+	return v.Storage
+	return nil
```
Comment (rinor) [inline provider/linode/linode.go:39]:
```suggestion
```

**Review by rinor — CHANGES_REQUESTED** [review]

**Comment by rinor** [comment]
Just curious, any reason not to use the official Linode go client? Now that Akamai bought Linode, it might be safer to follow that client.

**Comment by eyberg** [comment]
no other reason other than I didn't know it existed, i'm ok if someone wanted to add the changes for it but it's not the first one that we added support w/out an official client for and generally prefer not to pull in 3rd party deps until there's a need (but in this sort of case it makes sense)

**Pull request merged**